apps/system/js/app_chrome.js

```diff
@@ -340,6 +373,67 @@
     window.dispatchEvent(new CustomEvent('global-search-request'));
   };
 
+  AppChrome.prototype.onClickSiteIcon = function onClickSiteIcon() {
+    this.pinningPref && this.pinningPref.get(function(enabled) {
+      if (!enabled || !this.app.isBrowser()) {
+        this.app.debug('Pinning the web disabled');
+        return;
+      }
+      this.setPinDialogCard();
+    }.bind(this));
+  };
+
+  AppChrome.prototype.setPinDialogCard = function ac_setPinDialogCard(url) {
+    var card = new GaiaPinCard();
+    card.title = this.title.textContent;
```

Comment:
Ah, I should have spotted this earlier. This should be the title of the page, not the site name which is now what is displayed in the browser chrome. I guess this.title.textContent is misleadingly named, it should probably be this.name.